### ISSUE TITLE
[FEATURE] - Remove Job from Scheduler Jobs when the Job is supposed to be launch only once.

### ISSUE THREAD

**Issue by dlaweb** [go-co-op/gocron#85]
### Is your feature request related to a problem? Please describe

I've have a Job that i would like to run only once at a specific date

```
job, err = scheduler.Every(1).StartAt(date).SetTag(tag).Do(task)
...
job.LimitRunsTo(1)

```
However when the Job has run i still have a reference to it in the scheduler `scheduler.Jobs()`
 
I have to manually remove this specific Job in my task when the Job is done.

`scheduler.RemoveJobByTag(tag)`

Because if i don't, and if i run a second Job after this, then the second job will never be launched.

### Describe the solution you'd like

When we choose to limit the job to be runned only once `( job.LimitRunsTo(1) )` and its execution has been done then remove this Job from the scheduler Jobs.

**Comment by JohnRoesler** [comment]
@dlaweb thanks for reporting this! 

> Because if i don't, and if i run a second Job after this, then the second job will never be launched.

Are you later adding jobs to the running scheduler? Just trying to understand how the 2nd job is added so we can look at where the logic is that's causing the limited job to run again.

**Comment by dlaweb** [comment]
_Are you later adding jobs to the running schedule_ : Yes

[1] My program runs the scheduler

```
scheduler.StartAsync()

job, err = scheduler.Every(1).StartAt(**date**).SetTag(tag).Do(task)
...
job.LimitRunsTo(1)
```

[2] I send a request in order to run a Job:

 Sending request with **date** ( today 17:45 ) ...

-> OK , correctly executed task

[3] I send a another request in order to run a second Job:

Sending request with **date** ( today 17:50 ) ...

-> X , never executed

**Comment by dlaweb** [comment]
> Are you later adding jobs to the running scheduler?

**Comment by dlaweb** [comment]
Sorry i've close this issue by mistake

**Comment by JohnRoesler** [comment]
I was able to recreate and I've found the bug with the 2nd job never running (code below). I will open a PR shortly to resolve! 

```golang
package main

import (
	"fmt"
	"strconv"
	"time"

	"github.com/go-co-op/gocron"
)

func taskWithParams(id string) {
	fmt.Println("task id : " + id)
}
func main() {
	s := gocron.NewScheduler(time.UTC)
	s.StartAsync()

	j1, _ := s.Every(1).StartAt(time.Now().Add(10*time.Second)).Do(taskWithParams, strconv.Itoa(1))
	j1.LimitRunsTo(1)

	j2, _ := s.Every(1).StartAt(time.Now().Add(20*time.Second)).Do(taskWithParams, strconv.Itoa(2))
	j2.LimitRunsTo(1)

	time.Sleep(30 * time.Second)
}
```

**Comment by dlaweb** [comment]
What do you think about removing a Job (after the job has been executed) from the scheduler when the Job is run only once.
Because if i run 100 single Job i'll have a scheduler.Jobs with 100 Jobs who will be outdated after their execution

**Comment by JohnRoesler** [comment]
That is an interesting idea. I could see cases where someone might want to increase the limit of a job.

e.g.

```golang
	s := gocron.NewScheduler(time.UTC)
	s.StartAsync()

	j1, _ := s.Every(1).StartAt(time.Now().Add(10*time.Second)).Do(taskWithParams, strconv.Itoa(1))
	j1.LimitRunsTo(1)

       // some time later

       j1.LimitRunsTo(2)
```

So, really I could see both being valuable. We could consider if there is a flag we could add that you can set when limiting the run to specify to remove it. e.g. `job.LimitRunsTo(1).RemoveAfterRun()`

**Comment by JohnRoesler** [comment]
The bug here has been fixed with release [v0.4.0](https://github.com/go-co-op/gocron/releases/tag/v0.4.0). 

I re-opened this issue to track the feature request element for supporting remove jobs upon a single run.
